### PR TITLE
Deposit Address Component

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [x] Feature
- [ ] Bug fix

### Linked issue

https://cybrid.atlassian.net/browse/CYB-1372

https://cybrid.atlassian.net/browse/CYB-1414

### Evidence

https://github.com/Cybrid-app/cybrid-sdk-ios/assets/7268597/a704a12c-cc7a-49f0-a4d9-622c8da61e68

### CODE REVIEW

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+        if let qrCodeImage {
+            qrCodeImageView.image = qrCodeImage
+        }
+
+        // -- QR Code Warning
+        let qrCodeWarning = UILabel()
+        qrCodeWarning.font = UIFont.systemFont(ofSize: 12, weight: .light)
+        qrCodeWarning.textAlignment = .center
+        qrCodeWarning.textColor = UIColor(hex: "#818181")
+        qrCodeWarning.text = "Send only \(assetCode) to this deposit address."
```
Comment (brent-cybrid) [inline CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Localize

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+        qrCodeWarning.below(qrCodeImageView, top: 5)
+        qrCodeWarning.centerHorizontal(parent: contentView)
+        qrCodeWarning.constraintHeight(14)
+
+        // -- Network Title
+        let networkTitle = UILabel()
+        networkTitle.font = UIFont.systemFont(ofSize: 13, weight: .light)
+        networkTitle.textAlignment = .left
+        networkTitle.textColor = UIColor(hex: "#818181")
+        networkTitle.text = "Network"
```
Comment (brent-cybrid) [inline CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Localize

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+        depositAddressCopyIcon.isUserInteractionEnabled = true
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(copyDepositAddressToClipboard(_:)))
+        depositAddressCopyIcon.addGestureRecognizer(gesture)
+
+        // -- Address Title
+        let addressTitle = UILabel()
+        addressTitle.font = UIFont.systemFont(ofSize: 13, weight: .light)
+        addressTitle.textAlignment = .left
+        addressTitle.textColor = UIColor(hex: "#818181")
+        addressTitle.text = "\(assetCode) Deposit Address"
```
Comment (brent-cybrid) [inline CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Here

File: CybridSDK/Components/DepositAddress/Modal/PaymentModal.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    private func setupViews() {
+
+        // -- Title
+        let assetCode = self.accountBalance.asset?.code ?? ""
+        let depositTitle = UILabel()
+        depositTitle.font = UIFont.systemFont(ofSize: 20, weight: .semibold)
+        depositTitle.textAlignment = .left
+        depositTitle.textColor = UIColor.black
+        depositTitle.text = "Deposit details for \(assetCode)"
```
Comment (brent-cybrid) [inline CybridSDK/Components/DepositAddress/Modal/PaymentModal.swift]: Localize

File: CybridSDK/Utilities/Localization/en.lproj/Localizable.strings [hunk 5]
```diff
@@ -84,6 +84,17 @@
...
 "cybrid.accounts.transfers.detail.close.button" = "Close";
 
+// Deposit Address Component
+"cybrid.deposit.address.in.accounts.trades.create.button" = "Get Deposit Address";
+"cybrid.deposit.address.loading.label.verifying" = "Verifying your deposit addresses";
+"cybrid.deposit.address.loading.label.getting" = "Getting your deposit address";
+"cybrid.deposit.address.loading.label.creating" = "Creating deposit address";
+"cybrid.deposit.address.content.deposit.address.title" = "Deposit Address";
+"cybrid.deposit.address.content.deposit.tag.title" = "Tag";
+"cybrid.deposit.address.content.warning" = "Warning: You can lose all of your coins, bla bla bla bla bla bla bla bla bla bla bla blabla bla bla bla bla blabla bla bla bla bla blabla bla bla bla bla blabla bla bla bla bla blabla bla bla bla";
```
Comment (brent-cybrid) [inline CybridSDK/Utilities/Localization/en.lproj/Localizable.strings]: We can't push to main with this message. Let's get a more professional message here, even if it isn't the final version of the text.

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift (hunk 1 above)
Comment (nalancer08) [reply CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Done!

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift (hunk 2 above)
Comment (nalancer08) [reply CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Done!

File: CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift (hunk 3 above)
Comment (nalancer08) [reply CybridSDK/Components/DepositAddress/DepositAddresView+Extensions.swift]: Done!

File: CybridSDK/Components/DepositAddress/Modal/PaymentModal.swift (hunk 4 above)
Comment (nalancer08) [reply CybridSDK/Components/DepositAddress/Modal/PaymentModal.swift]: Done!

File: CybridSDK/Utilities/Localization/en.lproj/Localizable.strings [hunk 6]
```diff
@@ -84,6 +84,29 @@
...
+"cybrid.deposit.address.content.deposit.amount.title" = "Amount";
+"cybrid.deposit.address.content.deposit.message.title" = "Message";
+"cybrid.deposit.address.content.warning" = "Warning: You can lose all of your coins";
+"cybrid.deposit.address.content.button" = "Generate payment code";
+"cybrid.deposit.address.modal.title" = "Deposit details for";
+"cybrid.deposit.address.modal.amount.title" = "Amount of";
+"cybrid.deposit.address.modal.message.title" = "Message for transaction";
+"cybrid.deposit.address.modal.message.placeholder" = "Optional message";
+"cybrid.deposit.address.modal.button" = "Generate";
+"cybrid.deposit.address.error" = "Something happened, please try again.";
```
Comment (brent-cybrid) [inline CybridSDK/Utilities/Localization/en.lproj/Localizable.strings]: `An error occurred, please try again.`

**Review by brent-cybrid — APPROVED** [review]

File: CybridSDK/Utilities/Localization/en.lproj/Localizable.strings (hunk 6 above)
Comment (nalancer08) [reply CybridSDK/Utilities/Localization/en.lproj/Localizable.strings]: Done!

**Pull request merged**